compiler/rustc_lexer/src/unescape.rs

```diff
@@ -356,8 +356,10 @@ where
         let start = src.len() - chars.as_str().len() - c.len_utf8();
         let res = match c {
             '\\' => {
-                match chars.clone().next() {
+                let mut chars_clone = chars.clone();
+                match chars_clone.next() {
                     Some('\n') => {
+                        chars = chars_clone;
```

Comment:
this is not the prettiest, would this work?

```suggestion
                match chars.as_str().as_bytes().get(1) {
                    Some(b'\n') => {
                        let _ = chars.next();
```

it's fewer instructions total in an isolated example, and to me also a bit clearer.

Reply:
ah it's `.get(0)` I guess? or just `.first()` (based on the [raw error logs](https://productionresultssa6.blob.core.windows.net/actions-results/d7533e59-bac1-435c-807a-0077a5feaefe/workflow-job-run-6d6b583f-b7ad-5dcd-7a7f-872294aa7e9d/logs/job/job-logs.txt?rsct=text%2Fplain&se=2025-02-20T17%3A51%3A11Z&sig=iqcIPpF2lYlBe%2B4w06k4o5n%2F%2BoGnGTVlB57%2BE387P7g%3D&ske=2025-02-21T05%3A22%3A08Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2025-02-20T17%3A22%3A08Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2025-01-05&sp=r&spr=https&sr=b&st=2025-02-20T17%3A41%3A06Z&sv=2025-01-05)